### ISSUE TITLE
oshmem tests fail with Open MPI 4.1.5

### ISSUE THREAD

**Issue by mrogowski** [mpi4py/shmem4py#12]
Process 0 passes the tests then exits with Segmentation fault:
https://github.com/mpi4py/shmem4py/actions/runs/4730192013/jobs/8393588547

**Comment by mrogowski** [comment]
Reproducible with `hello.py`:

```
[root@a94d5472488d demo]# oshrun -n 1 python -u hello.py 
[1681807068.002249] [a94d5472488d:428  :0]            sock.c:502  UCX  WARN  unable to read somaxconn value from /proc/sys/net/core/somaxconn file
Hello from PE 0 of 1
[a94d5472488d:428  :0:428] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x7f8bcc1893e0)
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
oshrun noticed that process rank 0 with PID 0 on node a94d5472488d exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```

**Comment by mrogowski** [comment]
https://github.com/open-mpi/ompi/issues/11604

**Comment by mrogowski** [comment]
It looks like the segmentation fault is fixed as of Open MPI 4.1.6rc1. Our tests pass, but I see:
```[1691508125.245492] [d8804359d5d0:27   :0]        mm_iface.c:781  UCX  ERROR mm_iface failed to allocate receive FIFO```
and
```[1691508115.594522] [235a036d42d4:34   :0]           mpool.c:54   UCX  WARN  object 0x55f761f04d40 was not returned to mpool ucp_rkeys```
in the output.

**Comment by mrogowski** [comment]
Upgraded to Open MPI 4.1.6. We still see `[1698614657.831463] [52aae94dcb24:27   :0]           mpool.c:54   UCX  WARN  object 0x55af469df200 was not returned to mpool ucp_rkeys`, but everything seems to be working fine. Closing this issue.
